azure/provider.go

```diff
@@ -25,6 +26,9 @@ func (p *Provider) Name() string { return "Azure" }
 // Meta returns the provider metadata.
 func (p *Provider) Meta() unused.Meta { return p.meta }
 
+// Id returns the subscription for this provider.
+func (p *Provider) Id() string { return p.subscription }
```

Comment:
Given that in `NewProvider` we assign this field the value of `client.SubscriptionID`, and that we are storing `client` within the struct, we could get rid of this field:
```suggestion
func (p *Provider) Id() string { return p.client.SubscriptionID }
```

WDYT?

Reply:
done - removed from there as well as the struct definition.